webapp/packages/core-administration/src/AdministrationScreen/ConfigurationWizard/ConfigurationWizardService.ts

```diff
@@ -73,34 +72,62 @@ export class ConfigurationWizardService {
     private administrationItemService: AdministrationItemService,
     private administrationScreenService: AdministrationScreenService,
     private notificationService: NotificationService,
+    private screenService: ScreenService,
+    private serverConfigResource: ServerConfigResource,
   ) {
     makeObservable(this, {
       steps: computed,
       stepsToFinish: computed,
-      finishedSteps: computed,
       currentStepIndex: computed,
       canFinish: computed,
       nextStep: computed,
       currentStep: computed,
     });
+
+    this.screenService.routeChange.addHandler(this.onRouteChange.bind(this));
+  }
+
+  private async onRouteChange() {
+    await this.serverConfigResource.load();
+
+    const isExistingPage =
+      isNotNullDefined(this.administrationScreenService?.activeScreen?.item) &&
+      this.isStepAvailable(this.administrationScreenService.activeScreen.item);
+    const isCurrentStepAvailable = isNotNullDefined(this.currentStep) && this.isStepAvailable(this.currentStep.name);
+
```

Comment:
seems like `AdministrationScreenService` is better place for this logic (it's already have logic with redirects and access handling)